src/ui/hooks/useWindowSize.ts

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from 'preact/hooks';
+
+/** width at which "desktop" is returned */
+const MOBILE_BREAKPOINT = 600;
+
+let resizeTimeout: number | undefined;
+
+function useWindowSize() {
+  const [windowSize, setWindowSize] = useState(window.innerWidth);
+
+  function getWidth() {
+    setWindowSize(window.innerWidth);
+  }
+
+  useEffect(() => {
+    window.addEventListener(
+      'resize',
+      () => {
+        resizeTimeout = window.requestAnimationFrame(getWidth);
```

Comment:
Little RAF utility for efficiently detecting window size.

Needed because in the mobile view, code lives on top of preview. But if we are increasing the window size, we need to set the tab back to **Preview** again because the “Code” tab won’t exist anymore. This is the only window size-based logic in JS, and it just makes sure the preview pane is never empty.